source/boundary_traction/initial_lithostatic_pressure.cc

```diff
@@ -219,20 +270,29 @@ namespace aspect
     template <int dim>
     Tensor<1,dim>
     InitialLithostaticPressure<dim>::
-    boundary_traction (const types::boundary_id /*boundary_indicator*/,
+    boundary_traction (const types::boundary_id boundary_indicator,
                        const Point<dim> &position,
                        const Tensor<1,dim> &normal_vector) const
     {
       // We want to set the normal component to the vertical boundary
       // to the lithostatic pressure, the rest of the traction
       // components are left set to zero. We get the lithostatic pressure
-      // from a linear interpolation of the calculated profile.
+      // from a linear interpolation of the calculated profile,
+      // unless we need the traction at the bottom boundary. In this
+      // case we return the deepest pressure computed, as this is the
+      // normal traction component we want to prescribe even though the
+      // depth might not equal the depth of the initial profile because the
+      // surface has been deformed.
       Tensor<1,dim> traction;
-      traction = -interpolate_pressure(position) * normal_vector;
+      if (boundary_indicator == this->get_geometry_model().translate_symbolic_boundary_name_to_id("bottom"))
+        traction = -pressure.back() * normal_vector;
+      else
+        traction = -interpolate_pressure(position) * normal_vector;
```

Comment:
Does this mean if traction is prescribed at side and bottom boundaries then there will be a pressure jump between the lowest point on the side boundary and the bottom boundary? That seems a bit unintuitive. What is the reasoning for always using the last point of the profile for the bottom boundary instead of computing it using the position?

Reply:
That's a good point; I'm biased by the particular boundary conditions we use for our simulations, which only have tractions on the bottom boundary.

The reasons I think I had for taking the deepest pressure in the pressure profile for the bottom boundary:
* We always want to prescribe the same pressure on the bottom boundary, independent of the location on the bottom boundary. 
* When mesh deformation occurs, the depth returned by the geometry model for a point on the bottom boundary might not actually correspond with the deepest depth in the pressure profile. Take the box geometry model for example: 
   * The depth returned is computed as depth(x,y) = extents[z] + initial_topography(x,y) - position[z], so with respect to the initial topography.
   * If the point on the bottom boundary where we initially compute the reference profile is at t1 higher than at t0, the depth returned by the geometry model is 0 (because the actual negative depth is cut off at zero).
   * If the same point is lower than its initial topography, the depth returned will be smaller than the depth of the bottom boundary in the pressure profile.
   * So for the bottom boundary, only taking the last pressure in the profile leads to getting the pressure we want to prescribe. 

This is making me think that instead we could also improve on the definition of depth with mesh deformation enabled, like Derek worked on in #5088 .